components/Subscribe/index.tsx

```diff
@@ -1,36 +1,73 @@
 /* eslint-disable @typescript-eslint/no-explicit-any */
-import { FC, useState } from "react";
+import { FC, useState, useEffect, useContext } from "react";
 import axios from "axios";
+import { Modal } from "../Modal";
+import UserInfoContext from "@/context/UserInfoContext";
+import { validateEmail } from "@/lib/helpers/validateEmail";
+import { SubscribeButton } from "./subscribeButton";
 
-export const Subscribe: FC= () => {
-	const [email, setEmail] = useState<any>("");
+export const Subscribe: FC = () => {
+	const [email, setEmail] = useState<string>("");
 	const [state, setState] = useState<any>("idle");
-	// const [errorMsg, setErrorMsg] = useState<any>(null);
+	const [isModalOpen, setIsModalOpen] = useState<boolean>(false);
+	const { userInfo } = useContext(UserInfoContext);
+	const isAuth = userInfo?.email;
+	const [errorMsg, setErrorMsg] = useState<string>("");
+
+
+	type SubscribeModal = boolean | null;
+
+	useEffect(() => {
+		if (!localStorage.getItem("subscribeOpen") || localStorage.getItem("subscribeOpen")) {
+			localStorage.setItem("subscribeOpen", JSON.stringify(true)); 
+			setIsModalOpen(true); 
+		}
+	}, []);
+
+	useEffect (() => {
+		const subscribeModal: SubscribeModal = JSON.parse(localStorage.getItem("subscribeOpen") as string); 
```

Comment:
maybe just add subscribeModal: boolean | null, that way no need to create another variable take up more space and better readability. But up to you whatever you decide if there is a reason you'd rather declare a type here I would love to know

Reply:
No special reason. Got it, I agree.